pkg/controller/job/success_policy_test.go

```diff
@@ -0,0 +1,349 @@
+			},
+		},
+		"criteria.succeededIndexes and criteria.succeededCount are specified; succeededIndexes met criteria; criteria overlaps succeededIndexes at last": {
+			completions:      10,
+			succeededIndexes: orderedIntervals{{1, 3}, {5, 7}},
+			successPolicy: &batch.SuccessPolicy{
+				Criteria: []batch.SuccessPolicyCriteria{{
+					SucceededIndexes: ptr.To("2-4,6-9"),
+					SucceededCount:   ptr.To[int32](4),
+				}},
+			},
+			wantMetSuccessPolicy: true,
+			wantMessage:          "Met criteria at index 0",
+		},
+		"criteria.succeededIndexes is specified; succeededIndexes didn't meet criteria; criteria completely overlaps succeededIndexes": {
+			completions:      10,
+			succeededIndexes: orderedIntervals{{1, 3}, {7, 8}},
+			successPolicy: &batch.SuccessPolicy{
+				Criteria: []batch.SuccessPolicyCriteria{{
+					SucceededIndexes: ptr.To("0-4,6-9"),
+				}},
+			},
+		},
+		"criteria.succeededIndexes and criteria.succeededCount are specified; succeededIndexes met criteria; criteria completely overlaps succeededIndexes": {
+			completions:      10,
+			succeededIndexes: orderedIntervals{{1, 3}, {7, 8}},
+			successPolicy: &batch.SuccessPolicy{
+				Criteria: []batch.SuccessPolicyCriteria{{
+					SucceededIndexes: ptr.To("0-4,6-9"),
+					SucceededCount:   ptr.To[int32](5),
+				}},
+			},
+			wantMetSuccessPolicy: true,
+			wantMessage:          "Met criteria at index 0",
+		},
+		"criteria.succeededIndexes is specified; succeededIndexes didn't meet criteria; criteria overlaps multiple succeededIndexes at last": {
+			completions:      10,
+			succeededIndexes: orderedIntervals{{1, 3}, {5, 9}},
+			successPolicy: &batch.SuccessPolicy{
+				Criteria: []batch.SuccessPolicyCriteria{{
+					SucceededIndexes: ptr.To("0-6,8-9"),
+				}},
+			},
+		},
+		"criteria.succeededIndexes and criteria.succeededCount are specified; succeededIndexes met criteria; criteria overlaps multiple succeededIndexes at last": {
+			completions:      10,
+			succeededIndexes: orderedIntervals{{1, 3}, {5, 9}},
+			successPolicy: &batch.SuccessPolicy{
+				Criteria: []batch.SuccessPolicyCriteria{{
+					SucceededIndexes: ptr.To("0-6,8-9"),
+					SucceededCount:   ptr.To[int32](7),
+				}},
+			},
+			wantMetSuccessPolicy: true,
+			wantMessage:          "Met criteria at index 0",
+		},
+		"criteria.succeededIndexes is specified; succeededIndexes didn't meet criteria; criteria overlaps succeededIndexes at first, and criteria equals succeededIndexes at last": {
+			completions:      10,
+			succeededIndexes: orderedIntervals{{1, 5}, {7, 10}},
+			successPolicy: &batch.SuccessPolicy{
+				Criteria: []batch.SuccessPolicyCriteria{{
+					SucceededIndexes: ptr.To("0-5,7-9"),
+				}},
+			},
+		},
+		"criteria.succeededIndexes and criteria.succeededCount are specified; succeededIndexes met criteria; criteria overlaps succeededIndexes at first, and criteria equals succeededIndexes at last": {
+			completions:      10,
+			succeededIndexes: orderedIntervals{{1, 5}, {7, 10}},
```

Comment:
add a case where a single succeeded orderedInterval matches multiple intervals from the criteria.

Reply:
@alculquicondor We already have "criteria overlaps succeededIndexes at first, and criteria equals succeededIndexes at last".
Doe it make sense?

Or, you mean another case?

Reply:
More like:
succeeded: `1-10`
criteria: `0-3,6-9`

Reply:
I see. I'll try to add it.

Reply:
Done.